api_tests/users/views/test_user_claim.py

```diff
@@ -0,0 +1,257 @@
+# -*- coding: utf-8 -*-
+import mock
+import pytest
+
+from api.base.settings.defaults import API_BASE
+from framework.auth.core import Auth
+from osf_tests.factories import (
+    AuthUserFactory,
+    ProjectFactory,
+    PreprintFactory,
+)
+
+@pytest.mark.django_db
+class TestClaimUser:
+
+    @pytest.fixture()
+    def referrer(self):
+        return AuthUserFactory()
+
+    @pytest.fixture()
+    def project(self, referrer):
+        return ProjectFactory(creator=referrer)
+
+    @pytest.fixture()
+    def preprint(self, referrer, project):
+        return PreprintFactory(creator=referrer, project=project)
+
+    @pytest.fixture()
+    def wrong_preprint(self, referrer):
+        return PreprintFactory(creator=referrer)
+
+    @pytest.fixture()
+    def unreg_user(self, referrer, project):
+        return project.add_unregistered_contributor(
+            'David Davidson',
+            'david@david.son',
+            auth=Auth(referrer),
+            save=True
+        )
+
+    @pytest.fixture()
+    def claimer(self):
+        return AuthUserFactory()
+
+    @pytest.fixture()
+    def url(self):
+        return '/{}users/{{}}/claim/'.format(API_BASE)
+
+    def payload(self, **kwargs):
+        payload = {
+            'data': {
+                'attributes': {}
+            }
+        }
+        _id = kwargs.pop('id', None)
+        if _id:
+            payload['data']['id'] = _id
+        if kwargs:
+            payload['data']['attributes'] = kwargs
+        return payload
+
+    def test_unacceptable_methods(self, app, url, claimer, unreg_user, project):
```

Comment:
Not sure if this test is necessary--it's mostly testing the framework code.

Reply:
This test is more of a canary against accidental edits than an assurance that DRF behaves as expected, but I can remove if you want.

Reply:
I don't feel strongly about removing it. As suggested by Corgibytes, it'd be good for us to be more judicious about adding integration tests, since they're relatively expensive (due to all the db work they need).

Perhaps we could come up with a way to test this without any db work.

```python
assert only_supports_methods(ClaimUser, ["POST"])
```

Certainly not a blocker to merging this, though.

Reply:
It's a pretty simple util, I'll add it